src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,172 @@
+        String dateBy = secondModifiedInput[1];
+        LocalDate localDateBy = LocalDate.parse(dateBy);
+        return new Deadline(description, localDateBy);
+    }
+
+    /**
+     * Handles the number input given from the user.
+     *
+     * @param   input the input from the user
+     * @return  the integer that the user specified
+     * @throws DukeException if there is an error parsing the input from the user
+     */
+    public static int parseInt(String input) throws DukeException {
+        if (input.length() == 0) {
+            throw new DukeException("Did you forget to specify which task to delete?");
+        }
+        try {
+            return Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            throw new DukeException("Come on. Give me a number instead! Like 1 or 3!");
+        }
+    }
+
+    /**
+     * Handles the case where user inputs a <code>Find</code>.
+     *
+     * @param input the input string from the user
+     * @return the string from the user
+     * @throws DukeException if there is an error in the input given from the user
+     */
+    public static String handleFind(String input) throws DukeException {
+        if (input.length() == 0) {
+            throw new DukeException("Did you forget to specify what you are looking for?");
+        } else {
+            return input;
+        }
+    }
+
+    /**
+     * Handles the reading of the input from the user.
+     *
+     * @param   input the input from the user
+     * @return  the <code>Command</code> corresponding to the input from the user
+     * @throws DukeException if there is an error parsing the input from the user
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] strArray = input.split(" ", 2);
+        String first = strArray[0];
+        String second = "";
+        if (strArray.length == 2) {
+            second = strArray[1];
+        }
+        Command cmd;
+        switch (first) {
+        case "bye":
+            cmd = new ByeCommand();
+            break;
+        case "list":
+            cmd = new ListCommand();
+            break;
+        case "mark":
+            cmd = new MarkDoneCommand(parseInt(second));
+            break;
+        case "unmark":
+            cmd = new MarkUndoneCommand(parseInt(second));
+            break;
+        case "todo":
+            cmd = new AddCommand(handleTodo(second));
+            break;
+        case "deadline":
+            cmd = new AddCommand(handleDeadline(second));
+            break;
+        case "event":
+            cmd = new AddCommand(handleEvent(second));
+            break;
+        case "delete":
+            cmd = new DeleteCommand(parseInt(second));
+            break;
+        case "on":
+            cmd = new OnDateCommand(LocalDate.parse(second));
+            break;
+        case "find":
+            cmd = new FindCommand(handleFind(second));
+            break;
+        default:
```

Comment:
I like that you followed the appropriate indentation rules for switch clause here.